operator/pkg/apis/operator/v1alpha1/defaults.go

```diff
@@ -243,6 +244,23 @@ func setDefaultsKarmadaWebhook(obj *KarmadaComponents) {
 	}
 }
 
+func setDefaultsKarmadaSearch(obj *KarmadaComponents) {
+	if obj.KarmadaSearch == nil {
```

Comment:
If karmadaSearch is nil, we should not init a `KarmadaSearch` pointer.

Reply:
done~